### ISSUE TITLE
Large stack arrays causing segfaults

### ISSUE THREAD

**Issue by tomdeakin** [lanl/SNAP#10]
In both the inner and outer convergence checking, the `df` array is allocated on the stack. When we build with the Arm 19.2 compiler, and if this array happens to be fairly large (say >1MB), we see a segmentation fault at runtime.

One fix might be to allocate `df` on the heap instead:
```
diff --git a/src/inner.f90 b/src/inner.f90
index fa0e8fe..63877a4 100644
--- a/src/inner.f90
+++ b/src/inner.f90
@@ -275,7 +275,8 @@ MODULE inner_module
 
     INTEGER(i_knd) :: n, g
 
-    REAL(r_knd), DIMENSION(nx,ny,nz,ng_per_thrd) :: df
+    REAL(r_knd), ALLOCATABLE, DIMENSION(:,:,:,:) :: df
+    ALLOCATE(df(nx,ny,nz,ng_per_thrd))
 !_______________________________________________________________________
 !
 !   Thread group loops for computing local difference (df) array.
@@ -330,6 +331,8 @@ MODULE inner_module
 !_______________________________________________________________________
 !_______________________________________________________________________
 
+  DEALLOCATE(df)
+
   END SUBROUTINE inner_conv
 
 
diff --git a/src/outer.f90 b/src/outer.f90
index 849cd95..1fc020b 100644
--- a/src/outer.f90
+++ b/src/outer.f90
@@ -299,7 +299,8 @@ MODULE outer_module
 
     REAL(r_knd) :: dft
 
-    REAL(r_knd), DIMENSION(nx,ny,nz,ng_per_thrd) :: df
+    REAL(r_knd), DIMENSION(:,:,:,:), ALLOCATABLE :: df
+    ALLOCATE(df(nx,ny,nz,ng_per_thrd))
 !_______________________________________________________________________
 !
 !   Thread to speed up computation of df by looping over groups. Rejoin
@@ -347,6 +348,8 @@ MODULE outer_module
       otrdone = .TRUE.
 
   !$OMP END MASTER
+
+  DEALLOCATE(df)
 !_______________________________________________________________________
 !_______________________________________________________________________
```

A better way would be to rewrite the loops as a reduction to the scalar `dfmxi(g)` to avoid needing this array entirely.

**Comment by zerr** [comment]
Thanks, Tom. I'll take a closer look soon.

Something like this I may wish to hold off on changing unless it additionally makes sense for the production code. Apologies ahead of time if it seems like this doesn't move quickly. Let's keep the thread open.

**Comment by zerr** [comment]
Resolved in v1.10... I hope. Closing issue. We can reopen if problem persists.
